fair_mango/metrics/superset.py

```diff
@@ -0,0 +1,318 @@
+from collections.abc import Sequence
+from itertools import chain, combinations
+
+import pandas as pd
+
+from fair_mango.dataset.dataset import Dataset
+from fair_mango.metrics.metrics import (
+    ConfusionMatrix,
+    DemographicParityDifference,
+    DemographicParityRatio,
+    DisparateImpactDifference,
+    DisparateImpactRatio,
+    EqualisedOddsDifference,
+    EqualisedOddsRatio,
+    EqualOpportunityDifference,
+    EqualOpportunityRatio,
+    FalsePositiveRateDifference,
+    FalsePositiveRateRatio,
+    PerformanceMetric,
+    SelectionRate,
+)
+
+
+def super_set_fairness_metrics(
```

Comment:
maybe make it as a class with only a "rank" method so that it is explicit we are calling rank behind the scene

Reply:
and detail why there it does not make sense to have a summary method

Reply:
The superset analysis functions are refactored to classes with methods `.rank()` and `.evaluate()` for the classes `SupersetFairnessMetrics` and `SupersetPerformanceMetrics` respectively.